app/src/controllers/filter/sagas.js

```diff
@@ -127,24 +131,32 @@ function* resetFilter({ payload: filterId }) {
   yield put(updateFilterSuccessAction(savedFilter));
 }
 
-function* createFilter({ payload: filter = {}, meta = {} }) {
+function* createFilter({ payload = {}, meta = {} }) {
+  const { hasFilterPermissions, isRemoveCustomFilter, ...filter } = payload;
   const launchFilters = yield select(launchFiltersSelector);
   const userId = yield select(userIdSelector);
   const lastNewFilterId = launchFilters.reduce(
     (acc, launchFilter) => (launchFilter.id < acc ? launchFilter.id : acc),
     0,
   );
+
+  const filterName = hasFilterPermissions
+    ? `${NEW_FILTER_PREFIX} ${-(lastNewFilterId - 1)}`
+    : CUSTOM_FILTER;
+
   const newFilter = {
     ...DEFAULT_FILTER,
     ...filter,
-    id: lastNewFilterId - 1,
-    name: filter.name
-      ? `${COPY_PREFIX} ${filter.name}`
-      : `${NEW_FILTER_PREFIX} ${-(lastNewFilterId - 1)}`,
+    id: hasFilterPermissions ? lastNewFilterId - 1 : CUSTOM_FILTER_ID,
+    name: filter.name ? `${COPY_PREFIX} ${filter.name}` : filterName,
     owner: userId,
   };
   yield put(addFilterAction(newFilter));
   yield put(changeActiveFilterAction(newFilter.id, meta));
+
+  if (isRemoveCustomFilter) {
```

Comment:
Can we rely only on `hasFilterPermissions` here?
Looks like without permissions for filter it is always expected to remove filter immediately after create, isn't it?